src/mbgl/renderer/painter_fill.cpp

```diff
@@ -108,6 +109,44 @@ void Painter::renderFill(FillBucket& bucket, const FillLayer& layer, const TileI
             // Draw the actual triangles into the color & stencil buffer.
             setDepthSublayer(0);
             bucket.drawElements(*patternShader, glObjectStore);
+
+            if (properties.antialias && stroke_color == fill_color) {
+                config.program = outlinePatternShader->getID();
+                outlinePatternShader->u_matrix = vtxMatrix;
+                config.lineWidth = 2.0f;
+
+                // Draw the entire line
+                outlinePatternShader->u_world = {{
+                    static_cast<float>(frame.framebufferSize[0]),
+                    static_cast<float>(frame.framebufferSize[1])
+                }};
+
+                outlinePatternShader->u_pattern_tl_a = (*posA).tl;
+                outlinePatternShader->u_pattern_br_a = (*posA).br;
+                outlinePatternShader->u_pattern_tl_b = (*posB).tl;
+                outlinePatternShader->u_pattern_br_b = (*posB).br;
+                outlinePatternShader->u_opacity = properties.opacity;
+                outlinePatternShader->u_image = 0;
+                outlinePatternShader->u_mix = properties.pattern.value.t;
+
+                outlinePatternShader->u_patternscale_a = {{
+                    1.0f / id.pixelsToTileUnits(imageSizeScaledA[0], state.getIntegerZoom()),
+                    1.0f / id.pixelsToTileUnits(imageSizeScaledB[1], state.getIntegerZoom())
+                }};
+                outlinePatternShader->u_patternscale_b = {{
+                    1.0f / id.pixelsToTileUnits(imageSizeScaledB[0], state.getIntegerZoom()),
+                    1.0f / id.pixelsToTileUnits(imageSizeScaledB[1], state.getIntegerZoom())
+                }};
+
+                outlinePatternShader->u_offset_a = std::array<float, 2>{{offsetAx, offsetAy}};
+                outlinePatternShader->u_offset_b = std::array<float, 2>{{offsetBx, offsetBy}};
+
+                config.activeTexture = GL_TEXTURE0;
+                spriteAtlas->bind(true, glObjectStore);
+
+                setDepthSublayer(0);
```

Comment:
I think this should be `setDepthSublayer(2);` like the non-pattern antialiasing call.

What this does is draws the outline at a further depth so that it doesn't get drawn on top of the inner part of the fill.